File: ImageFlow/ocaml/bridge/cicolor.c
#include <objc/objc.h>
#include <objc/objc-runtime.h>
#include <caml/mlvalues.h>
#include <caml/memory.h>

#include "objc.h"

static id CIColorClass = NULL;

CAMLprim value CIColor__colorWithRedGreenBlueAlpha(value r, value g, value b, value a) {
  CAMLparam4(r, g, b, a);
  static SEL sel = NULL;
  if (sel == NULL)
    sel = sel_registerName("colorWithRed:green:blue:alpha:");
  if (CIColorClass == NULL)
    CIColorClass = objc_getClass("CIColor");
  CAMLreturn(objc_wrap(objc_msgSend(CIColorClass, sel, Double_val(r), Double_val(g), Double_val(b), Double_val(a))));
}

File: ImageFlow/ocaml/bridge/cicolor.m
#import <caml/mlvalues.h>
#import <caml/memory.h>

#import <Foundation/Foundation.h>
#import <QuartzCore/QuartzCore.h>

#import "objc.h"

CAMLprim value CIColor__colorWithRedGreenBlueAlpha(value r, value g, value b, value a) {
  CAMLparam4(r, g, b, a);
  CAMLreturn(objc_wrap([CIColor colorWithRed:Double_val(r) green:Double_val(g) blue:Double_val(b) alpha:Double_val(a)]));
}
